lib/flapjack/data/notification_rule.rb

```diff
@@ -121,37 +121,39 @@ def to_jsonapi(opts = {})
         Flapjack.dump_json(json_data)
       end
 
-      # entity names match?
+      # If the rule has any entities, then one of them must match the event's entity
       def match_entity?(event_id)
-        return false unless @entities
-        @entities.include?(event_id.split(':').first)
+        return true unless @entities && @entities.length > 0
+        event_entity = event_id.split(':').first
+        @entities.include?(event_entity)
       end
 
-      # entity names match regex?
+      # If the rule has any regex_entities, then at least one of them must match the
+      # event's entity
       def match_regex_entities?(event_id)
-        return false unless @regex_entities && @regex_entities.length > 0
-        entity = event_id.split(':').first
+        return true unless @regex_entities && @regex_entities.length > 0
+        event_entity = event_id.split(':').first
         matches = 0
         @regex_entities.each do |regex_entity|
-          matches += 1 if /#{regex_entity}/ === entity
+          matches += 1 if /#{regex_entity}/ === event_entity
         end
-        matches >= @regex_entities.length
+        matches >= 1
       end
 
-      # tags match?
+      # If the rule has any tags, then they must all be present in the
+      # event's tags
       def match_tags?(event_tags)
-        return false unless @tags && @tags.length > 0
+        return true unless @tags && @tags.length > 0
```

Comment:
As the documentation says, if there are no tags then match_tags? should return true. I know that this is checked elsewhere, but it seems clearer to have it all in one place.